src/common/IComponent.hpp

```diff
@@ -231,6 +233,8 @@ class IComponent {
     */
    virtual ~IField() = default;
 
+   IField() = default;
```

Comment:
his seems useless, and even if you want a IField constructor it shall be protected and documented.